djangoprojects/provision.py

```diff
@@ -48,43 +52,44 @@ def create(droplet_name):
              "region": "nyc3",
              "size": "s-1vcpu-1gb",
              "image": "ubuntu-16-04-x64",
-             "ssh_keys": [23341984],
+             "ssh_keys": [fingerprint.hash_md5().replace("MD5:", "")],
              "backups": False,
              "ipv6": True,
              "user_data": None,
              "private_networking": None,
              "volumes": None,
              "tags": [
-                   "Name:"+droplet_name
+                   "name:"+droplet_name
                  ]
             }
-    api_url = '{0}droplets'.format(api_url_base)
-    response = requests.post(api_url, headers=headers, json=data)
-    time.sleep(25)  # delays for 25 seconds
-    if response.status_code == 202:
-        print("SUCCESSFULLY CREATED")
-        jsondata = json.loads(response.text)
-        response_id = (jsondata.get("droplet")).get("id")
-        response_status = (jsondata.get("droplet")).get("status")
-        data = {'id': response_id, 'status': response_status}
-    else:
-        print("FAILED TO CREATE")
-    return data
+    api_url = '{0}droplets'.format(API_URL_BASE)
+    response = requests.post(api_url, headers=HEADERS, json=data)
+    jsondata = response.json()
+    droplet = jsondata['droplet']
+    while True:
+        time.sleep(20)
+        response_status = droplet['status']
+        if response_status == 'new':
+            print("SUCCESSFULLY CREATED")
+            break
+        else:
+            print("FAILED TO CREATE")
 
 
 def delete(droplet_name):
-    api_droplet = api_url_base + 'droplets?tag_name=' + droplet_name
-    get_response = requests.get(api_droplet, headers=headers)
-    jsondata = json.loads(get_response.text)
-    try:
-        if droplet_name == "Name:" + (jsondata.get("droplets")[0]).get("name"):
-            response = requests.delete(api_droplet, headers=headers)
+    api_droplet = API_URL_BASE + 'droplets?tag_name=' + droplet_name
+    get_response = requests.get(api_droplet, headers=HEADERS)
+    jsondata = get_response.json()
```

Comment:
Fetch the id of droplet, if present, delete that. Deleting by tag won't return any message if droplet is missing. And this api has no expected body, so `.json` may not work